src/renderer/features/governance/aggregates/unlock.ts

```diff
@@ -0,0 +1,84 @@
+import { createEvent, restore, sample } from 'effector';
+import { or, spread } from 'patronum';
+
+import { UnlockChunkType } from '@shared/api/governance';
+import { Step } from '@shared/lib/utils';
+import { referendumModel } from '@/entities/governance';
+import { locksModel } from '../model/locks';
+import { networkSelectorModel } from '../model/networkSelector';
+import { unlockModel } from '../model/unlock/unlock';
+
+import { unlockConfirmAggregate } from './unlockConfirm';
+
+const flowStarted = createEvent();
+const flowFinished = createEvent();
+const stepChanged = createEvent<Step>();
+const unlockConfirm = createEvent();
+const txSaved = createEvent();
+
+const $step = restore<Step>(stepChanged, Step.NONE);
+
+sample({
+  clock: flowStarted,
+  fn: () => Step.INIT,
+  target: stepChanged,
+});
+
+sample({
+  clock: txSaved,
+  fn: () => Step.BASKET,
+  target: stepChanged,
+});
+
+sample({
+  clock: stepChanged,
+  target: $step,
+});
+
+sample({
+  clock: flowFinished,
+  fn: () => Step.NONE,
+  target: stepChanged,
+});
```

Comment:
maybe you should rearrange samples a bit?
```suggestion
sample({
  clock: flowFinished,
  fn: () => Step.NONE,
  target: stepChanged,
});

sample({
  clock: stepChanged,
  target: $step,
});
```

Reply:
we usually have `flowFinished` in the end since it's one of the latest actions that  can be done in the flow